src/module/actor/mixins/actor-conditions.js

```diff
@@ -66,14 +92,50 @@ export const ActorConditionsMixin = (superclass) => class extends superclass {
             if (conditionItem) {
                 const promise = this.deleteEmbeddedDocuments("Item", [conditionItem.id]);
                 promise.then(() => {
-                    const updateData = {};
-                    updateData[`data.conditions.${conditionName}`] = false;
-                    this.update(updateData).then(() => {
-                        Hooks.callAll("onActorSetCondition", {actor: this, item: conditionItem, conditionName: conditionName, enabled: enabled});
-                    });
+                    this._updateActor(conditionName, false);
+                    Hooks.callAll("onActorSetCondition", {actor: this, item: conditionItem, conditionName: conditionName, enabled: enabled});
                 });
+
                 return promise;
             }
         }
     }
+
+    /**
+     * Updates the actor data with the condition settings and then checks if Flat-Footed needs to be updated.
+     *
+     * @param {String} conditionName The name of the condition matching keys from config.js SFRPG.statusEffectIconMapping
+     * @param {Boolean} enabled If this value is true it enables the condition
+     * @private
+     * */
+    _updateActor(conditionName, enabled) {
```

Comment:
_updateActor can be made async, and return the promise with the .then concatenation, to allow the callback to be performed after _updateActor finishes.

Reply:
duh!